surefire-api/src/main/java/org/apache/maven/surefire/booter/SurefireReflector.java

```diff
@@ -160,7 +158,7 @@ public Object convertIfRunResult( Object result )
         /**
          * @param delegate a target
          */
-        public ClassLoaderProxy( Object delegate )
+        ClassLoaderProxy( Object delegate )
```

Comment:
why removing public constructor this is not backward compat???? if checkstyle complains about this so remove checkstyle usage as it's useless....

Reply:
This constructor wasn't used outside the package. Intellij also confirms it, so checkstyle reports it properly. Also keep in mind that tests passes. If it not being used, access can be safly lowered to package level.